docs/docs-beta/docs/guides/quality-testing/asset-checks.md

```diff
@@ -1,4 +1,64 @@
 ---
 title: "Test assets with Asset Checks"
 sidebar_position: 10
----
\ No newline at end of file
+---
+
+Asset checks in Dagster provide a way to define and execute different types of data quality checks on your data assets directly in Dagster.
+
+This guide covers the most common use cases for testing assets with asset checks and taking action based on the result.
+
+<details>
+<summary>Prerequisites</summary>
+- Familiarity with [Assets](/concepts/assets)
+</details>
+
+## Testing assets with a single asset check
+
+The example below defines a single asset check on an asset that fails if the `order_id` column of the asset contains a null value.
+
+In this example, the asset check will run after the asset has been materialized, to ensure the quality of its data.
+
+<CodeExample filePath="guides/data-assets/quality-testing/asset-checks/single-asset-check.py" language="python" title="Asset with a single asset check" />
+
+## Testing assets with multiple asset checks
+
+In most cases, checking the data quality of an asset will require multiple checks.
+
+The example below defines two asset checks using the `@multi_asset_check` decorator:
+
+- One check that fails if the `order_id` column of the asset contains a null value
+- Another check that fails if the `item_id` column of the asset contains a null value
+
+In this example, both asset checks will run in a single operation after the asset has been materialized.
+
+<CodeExample filePath="guides/data-assets/quality-testing/asset-checks/multiple-asset-checks.py" language="python" title="Asset with multiple asset checks" />
+
+## Defining asset checks using a factory pattern
```

Comment:
I'd title this section "Programmatically generating asset checks" or something along those lines. It's a bit of a nitpick, but in my mind the factory pattern is just a tool to achieve the goal of "generate lots of similar asset checks", and so someone with that goal in mind would be better served by a title oriented around that.